drivers/spi/spi_ll_stm32.c

```diff
@@ -103,9 +129,18 @@ static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
 		spi_context_update_tx(&data->ctx, 2, 1);
 	}
 
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	while (!LL_SPI_IsActiveFlag_RXP(spi)) {
+		/* NOP if buffer has less than a packet to read*/
+	}
+	if (LL_SPI_IsEnabled(spi) && LL_SPI_GetMode(spi)) {
```

Comment:
ditto

Reply:
Same response :
`LL_SPI_IsEnabled` return either 1 (true) or 0 (false)
`LL_SPI_GetMode(spi)` return either :
```
#define LL_SPI_MODE_MASTER                      (SPI_CFG2_MASTER)
#define LL_SPI_MODE_SLAVE                          (0x00000000UL)
```

I agree this might not be super pretty to rely on the slave to be 0, and the master >0, do you prefer : 
`if (LL_SPI_IsEnabled(spi) && (LL_SPI_GetMode(spi) ==  LL_SPI_MODE_MASTER)) {`